pkg/oadcrp/models.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright Gen Digital Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package oadcrp
+
+// RegisterClientRequest represents a request to register an OAuth2 client.
+type RegisterClientRequest struct {
+	ClientName              string   `json:"client_name,omitempty"`
+	ClientURI               string   `json:"client_uri,omitempty"`
+	RedirectURIs            []string `json:"redirect_uris,omitempty"`
+	GrantTypes              []string `json:"grant_types,omitempty"`
+	ResponseTypes           []string `json:"response_types,omitempty"`
+	Scope                   string   `json:"scope,omitempty"`
+	TokenEndpointAuthMethod string   `json:"token_endpoint_auth_method,omitempty"`
```

Comment:
I see policy uri & jwks missing here. 
I understand it is hard to support each and every request param, so you can add custom parameter here (of type interface{}), will be useful during interop to provide additional client request param that may come up.

Reply:
I've added the missing params. It was easier to add in each one individually instead of trying to generalize it + add custom marshaller + support it via gomobile. However, if we do eventually need to add custom parameters that are outside of the core spec, then I can add that later